Engines/Wine/Shortcuts/Wine/script.js

```diff
@@ -1,186 +1,197 @@
+     * @returns {WineShortcut} The WineShortcut object
+     */
+    category(category) {
+        this._category = category;
+        return this;
+    }
 
-/**
-* sets shortcut arguments
-* @param {array} args shortcut arguments
-* @returns {WineShortcut} WineShortcut object
-*/
-WineShortcut.prototype.arguments = function (args) {
-    this._arguments = args;
-    return this;
-}
+    /**
+     * Sets the shortcut description
+     *
+     * @param {string} description The shortcut description
+     * @returns {WineShortcut} The WineShortcut object
+     */
+    description(description) {
+        this._description = description;
+        return this;
+    }
 
-/**
-* sets executable which shall be used
-* @param {string} search executable name
-* @returns {WineShortcut} WineShortcut object
-*/
-WineShortcut.prototype.search = function (search) {
-    this._search = search;
-    return this;
-}
+    /**
+     * Sets the shortcut arguments
+     *
+     * @param {array} args The shortcut arguments
+     * @returns {WineShortcut} The WineShortcut object
+     */
+    arguments(args) {
+        this._arguments = args;
+        return this;
+    }
 
-/**
-* sets shortcut prefix
-* @param {string} prefix shortcut prefix
-* @returns {WineShortcut} WineShortcut object
-*/
-WineShortcut.prototype.prefix = function (prefix) {
-    this._prefix = prefix;
-    return this;
-}
+    /**
+     * Sets the executable which shall be used
+     *
+     * @param {string} search The executable name
+     * @returns {WineShortcut} The WineShortcut object
+     */
+    search(search) {
+        this._search = search;
+        return this;
+    }
 
-/**
-* sets the miniature for the shortcut
-* @param {string[]|URI} miniature
-* array which specifies the application of which the miniature shall be used
-* or
-* URI of the miniature
-* @returns {WineShortcut} WineShortcut object
-*/
-WineShortcut.prototype.miniature = function (miniature) {
-    if (isArray(miniature)) {
-        // application of miniature given
-        var application = this._appsManager.getApplication(miniature);
-        if (application != null && application.getMainMiniature().isPresent()) {
-            this._miniature = application.getMainMiniature().get();
-        }
-    } else {
-        // miniature URI given
-        this._miniature = miniature;
+    /**
+     * Sets the shortcut prefix
+     *
+     * @param {string} prefix The shortcut prefix
+     * @returns {WineShortcut} The WineShortcut object
+     */
+    prefix(prefix) {
+        this._prefix = prefix;
+        return this;
     }
 
-    return this;
-}
+    /**
+     * Sets the miniature for the shortcut
+     *
+     * @param {string[]|URI} miniature An array which specifies the application of which the miniature shall be used or URI of the miniature
+     * @returns {WineShortcut} The WineShortcut object
+     */
+    miniature(miniature) {
+        if (isArray(miniature)) {
```

Comment:
@plata @ImperatorS79 @qparis does someone of you know where the `isArray` function is defined? It looks strange to me that we can use a function which we neither defined nor imported...

Reply:
https://docs.oracle.com/javase/8/docs/jdk/api/nashorn/jdk/nashorn/api/scripting/JSObject.html#isArray--

Reply:
Maybe we should try https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray?

Reply:
Yes.